packages/service-library/src/servicelib/file_utils.py

```diff
@@ -33,9 +40,11 @@ async def remove_directory(
         await _shutil_rmtree(path, ignore_errors=ignore_errors)
 
 
-class AsyncStream(Protocol):
-    async def read(self, size: int = -1) -> bytes:
-        ...
+def get_temporary_path_name() -> Path:
```

Comment:
I still do not understand why to get a path name that actually does not exist but I believe this could be achieved as

```python
from path import Path
import tempfile
from uuid import uuid4

temp_path_that_still_does_not_exist = Path(tempfile.gettempdir()) / f"{uuid4()}"
```

Reply:
Dropping this implementation and using the aiofiles suggestion